core/swupdate.c

```diff
@@ -772,9 +772,14 @@ int main(int argc, char **argv)
 			break;
 #ifdef CONFIG_DOWNLOAD
 		case 'd':
-			(void)snprintf(dwloptions, sizeof(dwloptions), "%s %s", argv[0], optarg);
+			if (asprintf(&dwloptions,"%s %s", argv[0], optarg) ==
+			ENOMEM_ASPRINTF) {
+				ERROR("Error in user input. Sufficient amount of memory cannot be allocated.");
```

Comment:
@TomaszAIR The message should describe what failed and for which module.
E.g. "Cannot allocate memory for downloader options."